### ISSUE TITLE
Add more style options for text markers

### ISSUE THREAD

**Issue by KatharinaSick** [osmdroid/osmdroid#1603]
## Issue Type

[ ] Question
[ ] Bug
[x] Improvement
[ ] Build system related
[ ] Performance
[ ] Documentation

## Description and/or steps/code to reproduce the problem
The layouting options for markers with text labels are very limited. Only colors, font size and anchor can be changed. Therefore it would be great to add some more styling options like:

* custom drawable as background
* font family
* corner radius
* padding

### Version of osmdroid the issue relates to:
6.1.2

**Comment by KatharinaSick** [comment]
I'm working on this myself - PR will be ready within the next days :)

**Comment by monsieurtanuki** [comment]
@KatharinaSick Looks similar to the TODO list of [`SpeechBalloonOverlay`](https://github.com/osmdroid/osmdroid/blob/master/osmdroid-android/src/main/java/org/osmdroid/views/overlay/SpeechBalloonOverlay.java).
If possible and relevant, please upgrade `SpeechBalloonOverlay` instead of creating a similar class.

**Comment by KatharinaSick** [comment]
Thanks @monsieurtanuki ! Actually I was planning to improve the Marker.java class itself, but of course I can also adapt the class you mentioned.

**Comment by monsieurtanuki** [comment]
If I remember well `InfoWindow` is based on a `View`, and `SpeechBalloonOverlay` on an `Overlay`. Some details are in #1347.

I don't know what exactly you have in mind, but I would not code directly into `Marker` anyway.
If it's about a marker that is not a logo (as now) but that is a nice text, I think an improved version of `SpeechBalloonOverlay` would do the trick.

**Comment by KatharinaSick** [comment]
Perfect - thank you! :)

**Comment by farmazon3000** [comment]
What is the problem? AFAIK you can give any drawable (or vector) as marker icon and you can make any padding (just give it any xml layout you want)... maybe I get the problem you have wrong? Please explain.

**Comment by KatharinaSick** [comment]
Yes I found this option @farmazon3000 but I would like to have/implement a text overlay (like the `SpeechBalloonOverlay`) where I can set these things automatically and just pass the text. Further I tried to create a custom drawable class where I can pass text, but this drawable was not displayed on the map.   

Further (and please correct me if I understood something wrongly) when arguing like that you can give the same argument against all custom overlay implementations?

**Comment by farmazon3000** [comment]
So maybe I did get you wrong - what I'm saying nothing stops you from implementing overlay with custom view.
